### PR TITLE
Closes #1404: Push sidebar down and fix horizontal scroll problem.

### LINKED ISSUE #1404: Move sidebar below last full width element on desktop
## Motivation
I want to see the sidebar aligned below the last full-width element on a page (only when sidebar region is next to content, not on mobile)

### Is your feature request related to a problem? Please describe.
Follow up ticket to https://github.com/az-digital/az_quickstart/issues/1202

### PR DESCRIPTION
**Pull request by trackleft**
## Related Issue

#1404 

## How to test?
Any page with both a full-width element link a text on media or text on background paragraph, AND one or two sidebars should be affected by this PR.

In any browser you deem supportable, add a page with a full-width paragraph AND other paragraphs.  Then add a block or two to either side of the page.

The sidebar should appear beneath the last full-width element on the page.
The rows of all of the full width paragraphs should be the width of the page container.

## Types of changes

**Arizona Quickstart** (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [x] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

**Drupal core**
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

**Drupal contrib projects**
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js [hunk 1]
```diff
@@ -1,24 +1,63 @@
...
+      document.documentElement.style.setProperty(
+        "--full-width-right-distance",
+        `${negativeRightMargin}px`
+      );
+    },
+  };
+
+  document.addEventListener('DOMContentLoaded', () => {
+    Drupal.behaviors.azParagraphsPushSidebarDown.attach();
+    Drupal.behaviors.calculateScrollbarWidth.attach();
```
Comment (trackleft) [inline modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js:55]:
```suggestion
```

File: modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js [hunk 2]
```diff
@@ -1,24 +1,63 @@
...
+    Drupal.behaviors.calculateFullWidthNegativeMargins.attach();
+  });
+  window.addEventListener('resize', () => {
+    Drupal.behaviors.calculateScrollbarWidth.attach();
```
Comment (trackleft) [inline modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js:59]:
```suggestion
Drupal.behaviors.azParagraphsPushSidebarDown.attach();
```

File: modules/custom/az_paragraphs/js/az_paragraphs_full_width.js [hunk 3]
```diff
@@ -5,16 +5,48 @@
...
+  document.addEventListener('DOMContentLoaded', function () {
+    Drupal.behaviors.azParagraphsPushSidebarDown.attach();
+    Drupal.behaviors.calculateFullWidthNegativeMargins.attach();
+  });
+  window.addEventListener('resize', function () {
+    Drupal.behaviors.calculateFullWidthNegativeMargins.attach();
+    Drupal.behaviors.calculateScrollbarWidth.attach();
+    Drupal.behaviors.azParagraphsPushSidebarDown.attach();
+  });
+})(Drupal, this, this.document);
```
Comment (trackleft) [inline modules/custom/az_paragraphs/js/az_paragraphs_full_width.js:52]:
```suggestion
})(Drupal, this, this.document);

```

**Review by akslay — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]
When loading the Text on Media page on Probo, the sidebar is sometimes placed in the wrong place (usually over the "Box With Full-Width Video") in Firefox, Safari, and Chrome. However, when testing locally this issue seems to happen less often. Since that page is a special case, I don't believe that issue should block this PR, but we could keep it in mind going forward.

**Comment by trackleft** [comment]
> When loading the Text on Media page on Probo, the sidebar is sometimes placed in the wrong place (usually over the "Box With Full-Width Video") in Firefox, Safari, and Chrome. However, when testing locally this issue seems to happen less often. Since that page is a special case, I don't believe that issue should block this PR, but we could keep it in mind going forward.

Hmm I thought we had that fixed. Darn

**Comment by trackleft** [comment]
Follow-up issue https://github.com/az-digital/az_quickstart/issues/1462

**Review by tadean — APPROVED** [review]

**Pull request merged**